### PR TITLE
define distributed as a type

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

prevent typos in selecting distributed types...
in next one we can simplify all `self.use_ddp`,  `self.use_ddp_spawn`, etc. as they are equal
Note that this is just internal mapping, nothing changes for the end user

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/accelerators/accelerator_connector.py [hunk 1]
```diff
@@ -216,19 +218,19 @@ def set_distributed_mode(self, distributed_backend):
                 self.trainer.use_ddp = True
                 self.trainer.num_processes = self.trainer.num_gpus
 
-        elif distributed_backend == "ddp2":
+        elif self.trainer.distributed_backend == BackendType.DDP2:
             # do nothing if num_gpus == 0
             if self.trainer.num_gpus >= 1:
                 self.trainer.use_ddp2 = True
-        elif distributed_backend == "ddp_cpu":
+        elif self.trainer.distributed_backend == BackendType.DDP2:
```
Comment (rohitgr7) [inline pytorch_lightning/accelerators/accelerator_connector.py:225]:
```suggestion
        elif self.trainer.distributed_backend == BackendType.DDP_CPU:
```

File: tests/models/test_tpu.py [hunk 2]
```diff
@@ -238,7 +239,7 @@ def test_exception_when_no_tpu_found(tmpdir):
 @pytest.mark.parametrize('tpu_cores', [1, 8, [1]])
 def test_distributed_backend_set_when_using_tpu(tmpdir, tpu_cores):
     """Test if distributed_backend is set to `tpu` when tpu_cores is not None"""
-    assert Trainer(tpu_cores=tpu_cores).distributed_backend == 'tpu'
+    assert Trainer(tpu_cores=tpu_cores).distributed_backend == BackendType.TPu
```
Comment (rohitgr7) [inline tests/models/test_tpu.py:242]:
```suggestion
    assert Trainer(tpu_cores=tpu_cores).distributed_backend == BackendType.TPU
```

File: pytorch_lightning/accelerators/base_backend.py [hunk 3]
```diff
@@ -175,3 +176,13 @@ def setup_optimizers(self, model):
...
         self.trainer.optimizer_frequencies = optimizer_frequencies
+
+
+class BackendType(Enum):
+    DP = 'dp'
+    DDP = 'ddp'
+    DDP2 = 'ddp2'
+    DDP_SPAWN = 'ddp_spawn'
+    DDP_CPU = 'ddp_cpu'
+    TPU = 'TPU'
```
Comment (rohitgr7) [inline pytorch_lightning/accelerators/base_backend.py:187]:
```suggestion
    TPU = 'tpu'
```

File: pytorch_lightning/accelerators/accelerator_connector.py [hunk 4]
```diff
@@ -194,18 +197,17 @@ def set_distributed_mode(self, distributed_backend):
                     ' Trainer(distributed_backend=dp) (or ddp, ddp2).'
                     ' Setting distributed_backend=ddp_spawn for you.'
```
Comment (rohitgr7) [inline pytorch_lightning/accelerators/accelerator_connector.py:198]: change the warnings too with `BackendType.mode`?

Comment (Borda) [reply pytorch_lightning/accelerators/accelerator_connector.py:198]: Trainer still works with strings...

**Comment by Borda** [comment]
@rohitgr7 great, thx for checking... 🐰

File: pytorch_lightning/accelerators/accelerator_connector.py (hunk 4 above)
Comment (rohitgr7) [reply pytorch_lightning/accelerators/accelerator_connector.py:198]: I meant using format strings and inject this new `BackendType` thing. It won't display `BackendType.mode` but its value. Just for consistency.

**Comment by rohitgr7** [comment]
How about doing the same for `amp_backend` and `amp_level` too??

**Comment by Borda** [comment]
> amp_backend

we already have amp_backend :]
https://github.com/PyTorchLightning/pytorch-lightning/blob/d5254ff9dfb67fba388de224a320f3a562561a80/pytorch_lightning/utilities/__init__.py#L25-L27

**Pull request merged**